table/update_schema.go

```diff
@@ -0,0 +1,834 @@
+package table
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/apache/iceberg-go"
+)
+
+// UpdateSchema accumulates operations and validates on Build/Commit
+type UpdateSchema struct {
+	txn          *Transaction
+	schema       *iceberg.Schema
+	lastColumnID int
+
+	operations []SchemaOperation
+	errors     []error // Collect validation errors
+
+	allowIncompatibleChanges bool
+	identifierFields         map[string]struct{}
+	caseSensitive            bool
+}
+
+// Operation represents a deferred schema operation
+type SchemaOperation interface {
+	Apply(builder *schemaBuilder) error
+	Validate(schema *iceberg.Schema, settings *validationSettings) error
+	String() string
+}
```

Comment:
since these operate only on internal types, why are we bothering to export this type?